airbyte-webapp/src/components/EntityTable/components/StreamStatusCell.tsx

```diff
@@ -0,0 +1,178 @@
+import { CellContext } from "@tanstack/react-table";
+import classNames from "classnames";
+import dayjs from "dayjs";
+
+import { Tooltip } from "components/ui/Tooltip";
+
+import {
+  WebBackendConnectionRead,
+  AirbyteStreamConfiguration,
+  ConnectionScheduleData,
+  ConnectionScheduleType,
+  JobStatus,
+  ConnectionStatus,
+} from "core/request/AirbyteClient";
+import { useGetConnection } from "hooks/services/useConnectionHook";
+
+import styles from "./StreamStatusCell.module.scss";
+import { Checkmark } from "./StreamStatusIcons/Checkmark";
+import { Error } from "./StreamStatusIcons/Error";
+import { Inactive } from "./StreamStatusIcons/Inactive";
+import { Late } from "./StreamStatusIcons/Late";
+import { Syncing } from "./StreamStatusIcons/Syncing";
+import { AirbyteStreamAndConfiguration } from "../../../core/request/AirbyteClient";
+import { ConnectionTableDataItem } from "../types";
+
+// TODO: When we have Actionable Errors, uncomment
+type StatusType = /* "Action Required" | */ "On Track" | "Disabled" | "Error" | "Behind";
```

Comment:
I know there was some discussion about enum vs types since the strings are type-enforced. I'm not the biggest fan of enums but curious what @edmundito thinks.

Reply:
After looking at this, I would argue for an enum (const enum specifically) since these statuses are representative of the data and as @dizel852 pointed out in another comment it would be easier to track. The values should be camelCase (or `snake_case` if we expect these to come from the backend int he future) but one advantage of camelCase is that they can be applied to i18n strings more easily.

Reply:
I would also name this `StreamStatus` or `StreamStatusType` to make search easier.